pkg/lib/authenticationflow/declarative/data_oauth.go

```diff
@@ -5,12 +5,21 @@ import (
 	"github.com/authgear/authgear-server/pkg/lib/oauthrelyingparty/wechat"
 )
 
+type OAuthProviderStatus string
+
+const (
+	OAuthProviderStatusActive   OAuthProviderStatus = "active"
+	OAuthProviderStatusInactive OAuthProviderStatus = "inactive"
+	// TODO(tung): Add demo status
+)
+
```

Comment:
Please also update the spec documentation.

> - status: The status about this option. It has the following valid values.
>   - active: The OAuth provider has credentials configured and it is usable.
>   - missing_credentials: The OAuth provider is missing credentials and it is NOT usable.
>   - using_demo_credentials: The OAuth provider is using a shared credentials for demo purpose. It is NOT for production use.

- Rename `inactive` to `missing_credentials`
- Use `using_demo_credentials` for the variant we has not yet introduce.

Reply:
@louischan-oursky I am ok with the renaming. But I also suggest to rename `disabled` of the oauth config to `missing_credential_allowed`, so we can relate them. What do you think?

Reply:
> But I also suggest to rename disabled of the oauth config to missing_credential_allowed, so we can relate them. What do you think?

OK.